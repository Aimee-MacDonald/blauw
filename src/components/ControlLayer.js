import React from 'react'
import styled from 'styled-components'

import Navigation from './controls/Navigation'
import Toolbox from './controls/Toolbox'

export default class ControlLayer extends React.Component{
  constructor(props){
    super(props)
  }

  render(){
    return(
      <StyledControlLayer>
        <Navigation/>
        <Toolbox/>
      </StyledControlLayer>
    )
  }
}

const StyledControlLayer = styled.div`
  position: fixed;
  z-index: 1;
  width: 100%;
  height: 0;
  overflow: visible;
  display: flex;
  flex-flow: nowrap row;
  justify-content: space-between;
`