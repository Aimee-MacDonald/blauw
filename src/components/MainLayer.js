import React from 'react'
import styled from 'styled-components'
import {connect} from 'react-redux'

import BookingSheet from './mains/BookingSheet'
import Checkout from './mains/Checkout'

const MainLayer = props => (
  <StyledMainLayer>
    {props.bookingSheet.open && <BookingSheet/>}
    {props.checkout.open && <Checkout/>}
  </StyledMainLayer>
)

const StyledMainLayer = styled.div`
  background-color: purple;
  position: fixed;
  z-index: 0;
  width: 100%;
  height: 100%;
`

const mapStateToProps = ({mains}) => ({
  bookingSheet: mains.bookingSheet,
  checkout: mains.checkout
})

export default connect(mapStateToProps)(MainLayer)