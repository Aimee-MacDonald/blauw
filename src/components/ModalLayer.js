import React from 'react'
import styled from 'styled-components'
import {connect} from 'react-redux'

import TempModal from './modals/TempModal'
import CreateBooking from './modals/CreateBooking'

const ModalLayer = props => (
  <StyledModalLayer {...props}>
    <CreateBooking />
  </StyledModalLayer>
)

const StyledModalLayer = styled.div`
  background-color: ${props => props.modalActive ? 'hotpink' : 'transparent'};
  position: fixed;
  z-index: 2;
  width: 100%;
  height: ${props => props.modalActive ? '100%' : '0'};
  overflow: hidden;
  transition: 0.4s;
  display: flex;
  justify-content: center;
  align-items: center;
`

const mapStateToProps = ({modal}) => ({
  modalActive: !!modal.activeModal
})

export default connect(mapStateToProps)(ModalLayer)