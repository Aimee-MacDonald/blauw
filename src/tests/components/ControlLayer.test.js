import React from 'react'
import {shallow} from 'enzyme'

import {ControlLayer} from '../../components/ControlLayer'

test('Render default Control Layer', () => {
  const wrapper = shallow(<ControlLayer/>)
  expect(wrapper.getElement()).toMatchSnapshot()
})