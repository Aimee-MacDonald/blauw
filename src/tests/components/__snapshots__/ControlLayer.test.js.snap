// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render default Control Layer 1`] = `
<styled.div>
  <Memo(Connect(StatusBar)) />
  <Memo(Connect(Navigation)) />
  <Memo(Connect(Toolbox)) />
</styled.div>
`;
